File: cinema-available-units-service/src/model/Movie.js
const db = require('../db')
const Sequelize = require('sequelize')

const Movie = db.define('movie', {
    id: {
        type: Sequelize.UUID,
        primaryKey: true
    },
    title: {
        type: Sequelize.STRING(50),
        allowNull: false
    },
    releaseYear: {
        type: Sequelize.INTEGER,
        allowNull: false
    }
})

module.exports = Movie

File: cinema-movies-service/src/main/java/com/abdul/cinemamoviesservice/model/Movie.java
package com.abdul.cinemamoviesservice.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.With;

import java.time.Year;
import java.util.UUID;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Movie {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    @Column(length = 50, nullable = false)
    private String title;
    private String description;

    @Column(nullable = false)
    private Year releaseYear;

    private Boolean inTheater;


}
